applications/NXstress.nxdl.xml

```diff
@@ -0,0 +1,1115 @@
+                    <dimensions rank="2">
+                        <dim index="1" value="n_D" />
+                        <dim index="2" value="n_X" />
+                    </dimensions>
+                    <attribute name="interpretation" type="NX_CHAR">
+                        <enumeration>
+                            <item value="spectrum" />
+                        </enumeration>
+                    </attribute>
+                </field>
+                <field name="residuals" type="NX_NUMBER" minOccurs="0" maxOccurs="1">
+                    <doc>Difference between diffractogram and fit (auxiliary signal).</doc>
+                    <dimensions rank="2">
+                        <dim index="1" value="n_D" />
+                        <dim index="2" value="n_X" />
+                    </dimensions>
+                    <attribute name="interpretation" type="NX_CHAR">
+                        <enumeration>
+                            <item value="spectrum" />
+                        </enumeration>
+                    </attribute>
+                </field>
+            </group>
+        </group>
+
+        <group name="NOTES" nameType="any" type="NXnote" minOccurs="0" maxOccurs="1">
+            <doc>
+                User description of the data acquisitions.
+                A description of data analysis goes in the
+                *fig_log group*.
+            </doc>
+        </group>
+
+        <group name="peaks" type="NXreflections" minOccurs="1" maxOccurs="1">
+            <doc>
+                This group contains all diffraction peak parameters that could be needed for stress and strain calculations. 
+                These parameters are derived from :ref:`peak_parameters &lt;/NXstress/ENTRY/fit/peak_parameters-group&gt;` and additional metadata.
+            </doc>
+            <field name="h" type="NX_INT" units="NX_UNITLESS">
+                <doc>First Miller index.</doc>
+                <dimensions rank="1">
+                    <dim index="1" value="n_Peaks" />
+                </dimensions>
+            </field>
+            <field name="k" type="NX_INT" units="NX_UNITLESS">
+                <doc>Second Miller index.</doc>
+                <dimensions rank="1">
+                    <dim index="1" value="n_Peaks" />
+                </dimensions>
+            </field>
+            <field name="l" type="NX_INT" units="NX_UNITLESS">
+                <doc>Third Miller index.</doc>
+                <dimensions rank="1">
+                    <dim index="1" value="n_Peaks" />
+                </dimensions>
+            </field>
+            <field name="lattice" type="NX_CHAR" minOccurs="0" maxOccurs="1">
+                <doc>Crystal lattice systems (*cubic*, *hexagonal*, ...)</doc>
+                <dimensions rank="1">
+                    <dim index="1" value="n_Peaks" />
+                </dimensions>
+            </field>
+            <field name="space_group" type="NX_CHAR" minOccurs="0" maxOccurs="1">

```

Comment:
Similar fields in `NXsample_component`

Reply:
Previous reviews resulted in the choice of NXreflections are class for the group. Fields like `space_group` can be found in NXcrytal as well but this does not help us here.

Reply:
Too late to change...